### ISSUE TITLE
Provide a faster alternative to UUID

### ISSUE THREAD

**Issue by Stebalien** [libp2p/go-libp2p-loggables#12]
UUIDs are slow. They require allocation and a systemcall. We should be using an instance unique UUID plus a global counter:

```go
type EventId struct {
    Inst: UUID // we can rotate this if we overflow uint64.
    Ctr: uint64
}
```

**Comment by Kubuxu** [comment]
@Stebalien uuid lib we are using uses `crypt/rand` for uuid generation: https://github.com/satori/go.uuid/blob/master/uuid.go#L113-L117

**Comment by Stebalien** [comment]
Yep. On Linux, at least, that always makes a systemcall (although I guess it doesn't allocate, it uses a fixed size byte array). Systemcalls are expensive. For now, we aren't doing anything where this would make a difference. However, if we switch to an RPC-based API with suffucient buffering, we could theoretically get down to <1 syscall per RPC. At that point, generating a UUID becomes expensive.

**Comment by Kubuxu** [comment]
That is true, Read from `crypt/rand` uses `getrandom(2)`.
